ding_spt_statistics.module

```diff
@@ -3,3 +3,153 @@
  * @file
  * Implementation of hooks and basic functionality goes here.
  */
+
+/**
+ * Implements hook_menu().
+ */
+function ding_spt_statistics_menu() {
+  $items = array();
+
+  $items['spt/redirect'] = array(
+    'title' => 'Ding SPT redirect',
+    'description' => 'Count click and redirect to the external resource',
+    'type' => MENU_CALLBACK,
+    'access arguments' => array('access content'),
+    'page callback' => 'ding_spt_statistics_redirect',
+  );
+  $items['spt/statistics'] = array(
+    'title' => 'Ding SPT statistics',
+    'description' => 'Count click and redirect to the external resource',
+    'type' => MENU_CALLBACK,
+    'access arguments' => array('view spt statistics'),
+    'page callback' => 'ding_spt_statistics_display',
+  );
+
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function ding_spt_statistics_permission() {
+  return array(
+    'view spt statistics' => array(
+      'title' => t('View SPT statistics'),
+      'description' => t('View statistics for clicked external SPT resources.'),
+    ),
+  );
+}
+
+/**
+ * Count click and redirect to the resource.
+ */
+function ding_spt_statistics_redirect() {
+  // Register statistical data.
+  $hostname = !empty($_GET['hostname']) ? $_GET['hostname'] : FALSE;
```

Comment:
If you have more then 1 params, better use - https://api.drupal.org/api/drupal/includes%21common.inc/function/drupal_get_query_parameters/7.x